desc/objectives/linear_objectives.py

```diff
@@ -917,6 +922,436 @@ def target_arg(self):
         return "c_l"
 
 
+class FixElectronTemperature(_FixProfile):
+    """Fixes electron temperature profile coefficients.
+
+    Parameters
+    ----------
+    eq : Equilibrium, optional
+        Equilibrium that will be optimized to satisfy the Objective.
+    target : tuple, float, ndarray, optional
+        Target value(s) of the objective.
+        len(target) = len(weight) = len(modes). If None, uses profile coefficients.
+    weight : float, ndarray, optional
+        Weighting to apply to the Objective, relative to other Objectives.
+        len(target) = len(weight) = len(modes)
+    normalize : bool
+        Whether to compute the error in physical units or non-dimensionalize.
+    normalize_target : bool
+        Whether target should be normalized before comparing to computed values.
+        if `normalize` is `True` and the target is in physical units, this should also
+        be set to True.
+    profile : Profile, optional
+        Profile containing the radial modes to evaluate at.
+    indices : ndarray or bool, optional
+        indices of the Profile.params array to fix.
+        (e.g. indices corresponding to modes for a PowerSeriesProfile or indices
+        corresponding to knots for a SplineProfile).
+        Must have len(target) = len(weight) = len(modes).
+        If True/False uses all/none of the Profile.params indices.
+    name : str
+        Name of the objective function.
+
+    """
+
+    _scalar = False
+    _linear = True
+    _fixed = True
+    _units = "(eV)"
+    _print_value_fmt = "Fixed-electron-temperature profile error: {:10.3e} "
+
+    def __init__(
+        self,
+        eq=None,
+        target=None,
+        weight=1,
+        normalize=True,
+        normalize_target=True,
+        profile=None,
+        indices=True,
+        name="fixed-electron-temperaturet",
+    ):
+
+        super().__init__(
+            eq=eq,
+            target=target,
+            weight=weight,
+            normalize=normalize,
+            normalize_target=normalize_target,
+            profile=profile,
+            indices=indices,
+            name=name,
+        )
+
+    def build(self, eq, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if eq.electron_temperature is None:
+            raise RuntimeError(
+                "Attempting to fix electron temperature on an equilibrium with no "
+                + "electron temperature profile assigned"
+            )
+        profile = eq.electron_temperature
+        if self._normalize:
+            scales = compute_scaling_factors(eq)
+            self._normalization = scales["p"]
```

Comment:
is B^2 / 2 mu0 a good normalization for this? Te is probably O(1e3)-O(1e4) right? this pressure normalization for W7-X, to give context, is O(1e5). I guess that is in the ballpark but could be a bit high. Not saying to change but just commenting

Reply:
its not perfect. I'll update it to be `scales['p'] / (scales['n'] * elementary_charge)`, but since density is usually O(1/elementary charge) it comes out about the same. The main issue is we don't have any inherent scale for n and T other than what we commonly see in experiments, but that's not "universal" in the same way that pressure balance etc is.